src/Measurements.jl

```diff
@@ -95,7 +104,7 @@ function measurement(val::T, err::T) where {T<:AbstractFloat}
         return Measurement{T}(val, err, tag, Derivatives(newder, (val, err, tag)=>one(T)))
     end
 end
-measurement(val::Real, err::Real) = measurement(promote(float(val), float(err))...)
+measurement(val::Real, err::Real) = measurement(promote(val, err)...)
```

Comment:
maybe add an specific constructor `measurement(val::Num,err::Num)`? (and it's combinations (::Num,::Real) (::Real,::Num)

Reply:
Are we worried that, given `val` and `err` of different types,
the type they will be promoted to initially, will still require re-promotion,
and we want to avoid that?